controllers/testclustertopology_controller.go

```diff
@@ -115,7 +115,52 @@ func (r *TestClusterTopologyReconciler) Reconcile(req ctrl.Request) (ctrl.Result
 			// TODO: wait for the cluster be ready
 		}
 	case naglfarv1.ClusterTopologyStateReady:
-		// DO NOTHING
+		// first create
+		if ct.Status.PreServerConfigs == nil && ct.Status.PreVersion == nil {
+			ct.Status.PreServerConfigs = ct.Spec.TiDBCluster.ServerConfigs.DeepCopy()
+			ct.Status.PreVersion = ct.Spec.TiDBCluster.Version.DeepCopy()
+			if err := r.Status().Update(ctx, &ct); err != nil {
+				log.Error(err, "unable to update TestClusterTopology")
+				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
+			}
+			return ctrl.Result{}, nil
+		}
+		// no change
+		if !tiup.IsServerConfigModified(*ct.Status.PreServerConfigs, ct.Spec.TiDBCluster.ServerConfigs) {
+			log.Info("Cluster no change")
+			return ctrl.Result{}, nil
+		}
+		log.Info("Cluster is updating", "clusterName", ct.Name)
```

Comment:
cluster is namespace-scoped, so logs should attach the ns name.

Reply:
done